spyderlib/utils/introspection/rope_plugin.py

```diff
@@ -93,10 +93,11 @@ def get_completions(self, info):
             proposals = rope.contrib.codeassist.sorted_proposals(proposals)
             if DEBUG_EDITOR:
                 log_dt(LOG_FILENAME, "code_assist/sorted_proposals", t0)
-            return [proposal.name for proposal in proposals]
+            return [(proposal.name, proposal.type) for proposal in proposals]
         except Exception as _error:  #analysis:ignore
             if DEBUG_EDITOR:
                 log_last_error(LOG_FILENAME, "get_completion_list")
+        return []

```

Comment:
To make sure we always return a list (I wasn't sure what else might break if I didn't).